oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -117,13 +125,27 @@ private ComputeEngineCredentials(HttpTransportFactory transportFactory) {
    * @return new ComputeEngineCredentials
    */
   public static ComputeEngineCredentials create() {
-    return new ComputeEngineCredentials(null);
+    return new ComputeEngineCredentials(null, null);
+  }
+
+  /**
+   * Create a new ComputeEngineCredentials instance based on this one, but with scopes explicitly
+   * specified.
+   */
+  @Override
+  public ComputeEngineCredentials createScoped(Collection<String> scopes) {
+    return new ComputeEngineCredentials(scopes, this.transportFactory);
   }
 
   /** Refresh the access token by getting it from the GCE metadata server */
   @Override
   public AccessToken refreshAccessToken() throws IOException {
-    HttpResponse response = getMetadataResponse(getTokenServerEncodedUrl());
+    GenericUrl url = new GenericUrl(getTokenServerEncodedUrl());
+    if (scopes != null) {
+      url.set("scopes", String.join(",", scopes));
```

Comment:
Looks like String.join was added in Java 8 and this library still currently support Java 7.

Reply:
You can use Guava's Joiner in this class.